flow/layers/layer.h

```diff
@@ -223,15 +223,19 @@ class Layer {
 
     ~AutoCachePaint() { context_.inherited_opacity = sk_paint_.getAlphaf(); }
 
-    void setImageFilter(sk_sp<SkImageFilter> filter) {
-      sk_paint_.setImageFilter(filter);
-      dl_paint_.setImageFilter(DlImageFilter::From(filter));
+    void setImageFilter(const DlImageFilter* filter) {
+      if (!filter) {
```

Comment:
I don't think it would matter given how this is used, but should a null pointer perhaps result in us clearing the attribute of the paints (just in case for future uses)?

Reply:
fixed